### PR TITLE
Added GetProtocolInfoResponseMessage docs

### LINKED ISSUE #3281: Links to GetProtocolInfoResponseMessage are broken
See https://community.dataminer.services/question/get-parameter-name-from-id-within-automation-script/

### PR DESCRIPTION
**Pull request by PedroDebevere**
Fixes #3281

### CODE REVIEW

File: src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs [hunk 1]
```diff
@@ -1,12 +1,1127 @@
...
+		/// </summary>
+		/// <exception cref="ArgumentNullException"><paramref name="protocolInfo"/> is <see langword="null"/>.</exception>
+		protected GetProtocolInfoResponseMessage(GetProtocolInfoResponseMessage protocolInfo)
+		{
+			if (protocolInfo == null)
+				throw new ArgumentNullException("protocolInfo");
+		}
+
+		/// <summary>
+		/// Links all of the child parameter info object to this object.
```
Comment (MariekeGO) [inline src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs:34]: Something seems to be wrong with this sentence. Is this what is meant?
```suggestion
		/// Links all of the child parameter info objects to this object.
```

File: src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs [hunk 2]
```diff
@@ -1,12 +1,1127 @@
...
+			return ProtocolTypes.Contains(typeToCheck);
+		}
+
+		/// <summary>
+		/// The main connection name.
+		/// </summary>
+		public String MainConnectionName;
+
+		/// <summary>
+		/// Converts the specified type to <see cref="¨ProtocolType"/>.
```
Comment (MariekeGO) [inline src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs:413]: Is this "¨" a mistake?
```suggestion
		/// Converts the specified type to <see cref="ProtocolType"/>.
```

File: src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs [hunk 3]
```diff
@@ -1,12 +1,1127 @@
...
+		/// </summary>
+		/// <param name="unparsed">The protocol type.</param>
+		/// <returns>The corresponding <see cref="¨ProtocolType"/>.</returns>
```
Comment (MariekeGO) [inline src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs:416]:
```suggestion
		/// <returns>The corresponding <see cref="ProtocolType"/>.</returns>
```

File: src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs [hunk 4]
```diff
@@ -1,12 +1,1127 @@
...
+		/// </summary>
+		/// <returns>Information about which pages have been defined in the protocol, and which parameters are on each of these pages.</returns>
+		/// <remarks>Only considers the actual device protocol. Does not return page info for the parameters of the base protocol.</remarks>
+		public PageInfo[] FindPages()
+		{
+			return null;
+		}
+
+		/// <summary>
+		/// Filters and/or sorts a collection of parameters, given a combination of selection options. e.g. returns a collection with only the analog parameters if <see pref="options"/> is set to <see cref="ParameterFilterOptions.AnalogOnly"/>.
```
Comment (MariekeGO) [inline src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs]: The `<see pref="options"/>` is not visible in my test build, leading to a weird sentence (same for line 975 below):

image

Comment (PedroDebevere) [reply src/DataMiner/SLNetTypes/Responses/GetProtocolInfoResponseMessage.cs]: corrected to <paramref name="options"/>

**Review by MariekeGO — APPROVED** [review]

**Pull request merged**